nncf/common/tensor_statistics/collectors.py

```diff
@@ -12,9 +12,10 @@
 from abc import ABC
 from abc import abstractmethod
 from collections import deque
-from typing import List, Optional, Tuple
+from typing import Any, Deque, Dict, Iterable, List, Optional, Tuple, Union, cast
 
 import numpy as np
+from numpy import typing as npt
```

Comment:
please do not import typing as npt and use `np.typing` instead

Reply:
Initially, I did use `np.typing`. When I ran some torch tests, I received the following error on my local machine, hence changed it:

`ImportError while loading conftest 'C:\nncf\tests\torch\conftest.py'.
tests\torch\conftest.py:25: in <module>
    from nncf.common.quantization.structs import QuantizationScheme as QuantizationMode
nncf\__init__.py:41: in <module>
    from nncf.quantization import QuantizationPreset as QuantizationPreset
nncf\quantization\__init__.py:13: in <module>
    from nncf.quantization.quantize_model import compress_weights as compress_weights
nncf\quantization\quantize_model.py:35: in <module>
    from nncf.quantization.algorithms.hyperparameter_tuner.algorithm import HyperparameterTuner
nncf\quantization\algorithms\hyperparameter_tuner\algorithm.py:19: in <module>
    from nncf.common.factory import NNCFGraphFactory
nncf\common\factory.py:19: in <module>
    from nncf.common.tensor_statistics import aggregator
nncf\common\tensor_statistics\aggregator.py:22: in <module>
    from nncf.common.tensor_statistics.statistic_point import StatisticPointsContainer
nncf\common\tensor_statistics\statistic_point.py:17: in <module>
    from nncf.common.tensor_statistics.collectors import TensorStatisticCollectorBase
nncf\common\tensor_statistics\collectors.py:21: in <module>
    from nncf.common.tensor_statistics.reduction import get_per_channel_history
nncf\common\tensor_statistics\reduction.py:27: in <module>
    def split_into_channels(input_: np.typing.NDArray[Any], scale_shape: List[int]) -> List[np.typing.NDArray[Any]]:
env\Lib\site-packages\numpy\__init__.py:347: in __getattr__
    raise AttributeError("module {!r} has no attribute "
E   AttributeError: module 'numpy' has no attribute 'typing'`

Reply:
Yes, it is ok, but I believe current approach is ok as well

Reply:
> Yes, it is ok, but I believe current approach is ok as well

The precommit/common pipeline fails because of the current approach as well. Hence, I suggested `from numpy.typing import NDArray`.